client/src/components/UserSettingField.tsx

```diff
@@ -67,30 +116,87 @@ type EmailSettingFieldProps = withI18nProps & {
 
 const EmailSettingFieldWithoutI18n = (props: EmailSettingFieldProps) => {
   const { i18n, currentValue, onSubmit } = props;
-  const [value, setValue] = useState(currentValue);
+  const userContext = useContext(UserContext);
+  const { email: oldEmail, verified } = userContext.user as JustfixUser;
+  const [existingUserError, setExistingUserError] = useState(false);
+  const {
+    value: email,
+    error: emailError,
+    showError: showEmailError,
+    setError: setEmailError,
+    setShowError: setShowEmailError,
+    onChange: onChangeEmail,
+  } = useInput(oldEmail);
 
-  const handleValueChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    setValue(e.target.value);
-  };
+  const handleSubmit = async () => {
+    setExistingUserError(false);
+    setShowEmailError(false);
+
+    if (email === oldEmail) {
+      return;
+    }
+
+    if (!email || emailError) {
+      setEmailError(true);
+      setShowEmailError(true);
+      throw new Error("Email format error");
```

Comment:
where would you see these exceptions thrown? in the browser console? just confused about the utility of the exception here and in the `handleSubmit` below since we are also setting the error states accordingly

Reply:
ugh yeah this was an issue with how these email and pw fields are nested within the form wrapper component and the `OnSubmit` gets passed around a lot (from the components props, to the `handleSubmit` to the props of the wrapper, to be called in `handleSubmit` in that wrapper. And so I couldn't just `return` at those error steps without the rest of the wrapper's `handleSubmit` running, which included the closing of the edit dropdown. (see my other comment)
I think this whole setup with the wrapper makes this confusing, but short of rewriting all of that this was the only way I could think of to get that working. Agree that it's not ideal still though. Would be nice to refactor maybe once we have the new form component library elements.